streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -1088,4 +1090,68 @@ public void shouldCreateStateDirectoryForStatefulTopology() {
         final TaskId taskId = new TaskId(0, 0);
         assertTrue(new File(appDir, taskId.toString()).exists());
     }
+
+    @Test
+    public void shouldReturnIteratorForMultipleRecords() {
+        testDriver = new TopologyTestDriver(setupSourceSinkTopology(), config);
+
+        testDriver.pipeInput(consumerRecord1);
+        final byte[] key2 = "key2".getBytes();
+        final byte[] value2 = "value2".getBytes();
+        final long timestamp2 = 42L;
+        final ConsumerRecord<byte[], byte[]> consumerRecord2 = consumerRecordFactory.create(SOURCE_TOPIC_1, key2, value2, headers, timestamp2);
+        testDriver.pipeInput(consumerRecord2);
+
+        final Iterator<ProducerRecord<byte[], byte[]>> output = testDriver.iterateOutput(SINK_TOPIC_1).iterator();
+
+        final ProducerRecord outputRecord1 = output.next();
+
+        assertEquals(key1, outputRecord1.key());
+        assertEquals(value1, outputRecord1.value());
+        assertEquals(SINK_TOPIC_1, outputRecord1.topic());
+
```

Comment:
FYI: There is the static nested class `Record` in `TopologyTestDriverTest`, that can be used to compare records.

Reply:
I don't see how I could use that here, as far as I can see it is used to compare the records from the MockProcessor to the input records.
In my case I would have to compare input to output records which don't match on the topic name.
Or am I missing something?

Reply:
Sorry, you are right! My bad!